Meadowlark-js/packages/metaed-plugin-edfi-meadowlark/src/enhancer/JsonSchemaEnhancer.ts

```diff
@@ -0,0 +1,443 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+/* eslint-disable no-use-before-define */
+
+import {
+  getAllEntitiesOfType,
+  MetaEdEnvironment,
+  EnhancerResult,
+  ReferentialProperty,
+  EntityProperty,
+  CommonProperty,
+  TopLevelEntity,
+  StringProperty,
+  IntegerProperty,
+  ShortProperty,
+} from '@edfi/metaed-core';
+import { invariant } from 'ts-invariant';
+import type { EntityMeadowlarkData } from '../model/EntityMeadowlarkData';
+import type { EntityPropertyMeadowlarkData } from '../model/EntityPropertyMeadowlarkData';
+import {
+  newSchemaRoot,
+  NoSchemaProperty,
+  SchemaArray,
+  SchemaObject,
+  SchemaProperties,
+  SchemaProperty,
+  SchemaRoot,
+} from '../model/JsonSchema';
+import { PropertyModifier, prefixedName } from '../model/PropertyModifier';
+import { singularize, topLevelNameOnEntity } from '../Utility';
+
+const enhancerName = 'JsonSchemaEnhancer';
+
+// All descriptor documents have the same schema
+const descriptorSchema: SchemaRoot = {
+  ...newSchemaRoot(),
+  type: 'object',
+  title: 'EdFi.Descriptor',
+  description: 'An Ed-Fi Descriptor',
+  properties: {
+    namespace: {
+      type: 'string',
+      description: 'The descriptor namespace as a URI',
+    },
+    codeValue: {
+      type: 'string',
+      description: 'The descriptor code value',
+    },
+    shortDescription: {
+      type: 'string',
+      description: 'The descriptor short description',
+    },
+    description: {
+      type: 'string',
+      description: 'The descriptor description',
+    },
+  },
+  additionalProperties: false,
+  required: ['namespace', 'codeValue', 'shortDescription'],
+};
+
+// SchoolYearEnumeration is a hardcoded exception in the ODS/API
```

Comment:
This is our secret easter egg for anyone outside of Ed-Fi who tries to use MetaEd & Meadowlark to stand up an alternate data standard  & API without paying attention.

Reply:
Maybe I should Base64 encode the comment? 🤣🤣